metadata-ingestion/src/datahub/ingestion/source/superset.py

```diff
@@ -0,0 +1,274 @@
+import json
+from functools import lru_cache
+from typing import Iterable, Optional
+
+import dateutil.parser as dp
+import requests
+
+from datahub.configuration.common import ConfigModel
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.api.source import Source, SourceReport
+from datahub.ingestion.source.metadata_common import MetadataWorkUnit
+from datahub.metadata.com.linkedin.pegasus2avro.common import (
+    AuditStamp,
+    ChangeAuditStamps,
+)
+from datahub.metadata.com.linkedin.pegasus2avro.metadata.snapshot import (
+    ChartSnapshot,
+    DashboardSnapshot,
+)
+from datahub.metadata.com.linkedin.pegasus2avro.mxe import MetadataChangeEvent
+from datahub.metadata.schema_classes import ChartInfoClass, DashboardInfoClass
+
+PAGE_SIZE = 25
+
+
+class SupersetConfig(ConfigModel):
+    # See the Superset /security/login endpoint for details
+    # https://superset.apache.org/docs/rest-api
+    connect_uri: str = "localhost:8088"
+    username: Optional[str] = None
+    password: Optional[str] = None
+    provider: str = "db"
+    options: dict = {}
+
+
+def get_platform_from_sqlalchemy_uri(sqlalchemy_uri: str) -> str:
+    if sqlalchemy_uri.startswith("bigquery"):
+        return "bigquery"
+    if sqlalchemy_uri.startswith("druid"):
+        return "druid"
+    if sqlalchemy_uri.startswith("mssql"):
+        return "mssql"
+    if (
+        sqlalchemy_uri.startswith("jdbc:postgres:")
+        and sqlalchemy_uri.index("redshift.amazonaws") > 0
+    ):
+        return "redshift"
+    if sqlalchemy_uri.startswith("snowflake"):
+        return "snowflake"
+    if sqlalchemy_uri.startswith("presto"):
+        return "presto"
+    if sqlalchemy_uri.startswith("postgresql"):
+        return "postgres"
+    if sqlalchemy_uri.startswith("pinot"):
+        return "pinot"
+    if sqlalchemy_uri.startswith("oracle"):
+        return "oracle"
+    if sqlalchemy_uri.startswith("mysql"):
+        return "mysql"
+    if sqlalchemy_uri.startswith("mongodb"):
+        return "mongo"
+    if sqlalchemy_uri.startswith("hive"):
+        return "hive"
+    return "external"
+
+
+class SupersetSource(Source):
+    config: SupersetConfig
+    report: SourceReport
+    env = "PROD"
+    platform = "superset"
+
+    def __hash__(self):
+        return 0
```

Comment:
Why do we want it to be hashable?

Reply:
Good question - this was necessary to implement LRU cache, since the LRU cache needs to hash each param (including self)

Reply:
So if someone creates two instances of the SupersetSource, will they end up sharing a cache?

Maybe we could do this? (although strictly speaking, this isn't correct either - 
 https://stackoverflow.com/questions/55085963/using-an-objects-id-as-a-hash-value )
```suggestion
        return id(self)
```

Reply:
cool- committing that 👍